File: 228.summary-ranges.py
#
# @lc app=leetcode id=228 lang=python3
#
# [228] Summary Ranges
#

# @lc code=start
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if not nums: return []
        
        result = []
        start = nums[0]
        
        for i in range(1, len(nums) + 1):
            if i == len(nums) or nums[i] - nums[i - 1] > 1:
                end = nums[i - 1]
                if start == end:
                    result.append(str(start))
                else:
                    result.append(str(start) + "->" + str(end))
                    
                if (i < len(nums)):
                    start = nums[i]
                
        return result
            
        
# @lc code=end

File: 228.summary-ranges.ts
/*
 * @lc app=leetcode id=228 lang=typescript
 *
 * [228] Summary Ranges
 */

// @lc code=start
function summaryRanges(nums: number[]): string[] {
   const output: string[] = [];
   let currStr: string | null = null;
   for (let i = 0; i < nums.length; i++) {
      if (nums[i] === nums[i + 1] || nums[i + 1] - nums[i] !== 1) {
         if (currStr === null) {
            output.push(nums[i].toString());
         } else {
            currStr += nums[i].toString();
            output.push(currStr);
            currStr = null;
         }
      } else {
         if (currStr === null) {
            currStr = nums[i].toString() + '->';
         }
      }
   }

   return output;
}
// @lc code=end
